core/test-runtime/src/lib.rs

```diff
@@ -591,6 +608,39 @@ cfg_if! {
 					(0..10000).fold(0, |p, i| p + benchmark_add_one(i))
 				}
 
+				fn returns_mutable_static() -> u64 {
+					unsafe {
+						MUTABLE_STATIC += 1;
+						MUTABLE_STATIC
+					}
+				}
+
+				fn allocates_stack_alot(trap: bool) {
+					// Allocate a stack frame that is approx. 75% of the stack (assuming it is 1MB).
+					// This will just decrease (stacks in wasm32-u-u grow downwards) the stack
+					// pointer. This won't trap on the current compilers.
+					let mut data = [0u8; 1024 * 768];
+
+					// Then make sure we actually write something to it.
+					//
+					// If:
+					// 1. the stack area is placed at the beginning of the linear memory space, and
+					// 2. the stack pointer points to out-of-bounds area, and
+					// 3. a write is performed around the current stack pointer.
+					//
+					// then a trap should happen.
+					//
+					for (i, v) in data.iter_mut().enumerate() {
```

Comment:
Are we sure that the compiler does not optimizes this away? Maybe also return the result of this function?

Reply:
Good point! I actually tried to do that with `set_storage` but it legitamately runs out of stack space in that case. It would be great if we could use `data.as_ptr()` somewhere where optimizer couldn't get that it is being tricked! Do you have an idea how?

Reply:
Could we not just call `to_vec()` and return a `Vec`?

Reply:
Let me try! Actually now I don't see what this could be a problem and if there is then it is most likely worth attention!

Reply:
Found it! Turns out that the heap is only 1mb and can't fit the allocation of 786432 bytes